zerver/tests/test_message_send.py

```diff
@@ -1574,6 +1575,40 @@ def test_not_too_many_queries(self) -> None:
                 body=content,
             )
 
+        do_change_user_setting(
+            user_profile=user,
+            setting_name="automatically_follow_topics_where_mentioned",
+            setting_value=True,
+            acting_user=None,
+        )
+        # There will be an increase in the query count of 5 while sending
+        # a message with a mention to a topic if visibility policy for the
+        # mentioned user is other than FOLLOWED.
+        # 1 to get the user_id of the mentioned user + 1 to check if the topic
+        # is already followed + 3 queries to follow the topic.
+        flush_per_request_caches()
+        with self.assert_database_query_count(22):
+            check_send_stream_message(
+                sender=sender,
+                client=sending_client,
+                stream_name=stream_name,
+                topic="topic 2",
+                body="@**" + user.full_name + "**",
+            )
+        # If the topic is already FOLLOWED, there will be an increase in the query
+        # count of 2.
+        # 1 to get the user_id of the mentioned user + 1 to check if the topic is
+        # already followed.
+        flush_per_request_caches()
+        with self.assert_database_query_count(19):
+            check_send_stream_message(
+                sender=sender,
+                client=sending_client,
+                stream_name=stream_name,
+                topic="topic 2",
+                body="@**" + user.full_name + "**",
+            )
```

Comment:
We should make sure there's at least one test covering the wildcard mention case for this logic.

Reply:
Added a test for stream wildcard mention.